### PR TITLE
Add the remote-dev command

### PR DESCRIPTION
**Pull request by stuartwdouglas**
This allows for remote container based development using a websocket connection.

### CODE REVIEW

**Comment by stuartwdouglas** [comment]
This needs some tests and docs before it is properly ready to be merged, but just opened a PR here for testing.

File: extensions/undertow-websockets/deployment/src/main/java/io/quarkus/undertow/websockets/deployment/HotReplacementWebsocketHandler.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
```
Comment (Sanne) [inline extensions/undertow-websockets/deployment/src/main/java/io/quarkus/undertow/websockets/deployment/HotReplacementWebsocketHandler.java]: you seem to have a random variety of headers. In particular this one mentionging LGPL is probably not OK :)

I think would be best if you could delete them all?

**Review by Sanne — DISMISSED** [review]
the LGPL headers need to be removed.

Secondarily: should we wait for Alexey's big changes first?

File: extensions/undertow-websockets/deployment/src/main/java/io/quarkus/undertow/websockets/deployment/HotReplacementWebsocketHandler.java (hunk 1 above)
Comment (stuartwdouglas) [reply extensions/undertow-websockets/deployment/src/main/java/io/quarkus/undertow/websockets/deployment/HotReplacementWebsocketHandler.java]: Oops, this was from some old code I had lying around

**Comment by Sanne** [comment]
> This needs some tests and docs before it is properly ready to be merged, but just opened a PR here for testing.

@stuartwdouglas are you happy with "tests and docs" aspect?

**Review by Sanne — APPROVED** [review]
Looking good. I'd merge it but @stuartwdouglas seemed to prefer to hold until he had some more tests/docs?

**Comment by stuartwdouglas** [comment]
I added tests and docs yesterday, so it should be all good

**Comment by vorburger** [comment]
> needs some tests and docs before it is properly ready to be merged
> @stuartwdouglas are you happy with "tests and docs" aspect?

would (my) help to test and contrib document re. this be welcome, or not needed anymore?

**Pull request merged**